lib/backup/Backup.ts

```diff
@@ -0,0 +1,179 @@
+
+  public start = async (args: { [argName: string]: any }) => {
+    await this.config.load(args);
+
+    this.backupDir = args.backupdir || this.getDefaultBackupDir();
+
+    this.logger = new Logger({
+      context: Context.Backup,
+      level: this.config.loglevel,
+      filename: this.config.logpath,
+      instanceId: this.config.instanceid,
+      dateFormat: this.config.logdateformat,
+    });
+
+    if (!fs.existsSync(this.backupDir)) {
+      try {
+        fs.mkdirSync(this.backupDir);
+      } catch (error) {
+        this.logger.error(`Could not create backup directory: ${error}`);
+        return;
+      }
+    }
+
+    try {
+      await this.startLndSubscriptions();
+    } catch (error) {
+      this.logger.error(`Could not connect to LNDs: ${error}`);
+    }
+
+    // Start the Raiden database filewatcher
+    if (args.raiden) {
+      this.startFilewatcher('raiden', args.raiden.database);
+    } else {
+      this.logFileNotSpecified('raiden');
+    }
+
+    // Start the XUD database filewatcher
+    if (args.raiden) {
+      this.startFilewatcher('xud', args.dbpath);
+    } else {
+      this.logFileNotSpecified('xud');
+    }
+
+    this.logger.info('Started backup daemon');
+  }
+
+  public stop = async () => {
+    this.fileWatchers.forEach((watcher) => {
+      watcher.close();
+    });
+
+    for (const lndClient of this.lndClients) {
+      await lndClient.close();
+    }
+  }
+
+  private startLndSubscriptions = async () => {
+    // Start the LND SCB subscriptions
+    for (const currency in this.config.lnd) {
+      const config = this.config.lnd[currency]!;
+
+      // Ignore the LND client if it is disabled or not configured
+      if (!config.disable && Object.entries(config).length !== 0) {
+        const lndClient = new LndClient({
+          config,
+          currency,
+          logger: Logger.DISABLED_LOGGER,
+        });
+
+        this.lndClients.push(lndClient);
+
+        await lndClient.init();
+
+        this.logger.verbose(`Writing initial ${lndClient.currency} LND channel backup`);
+
+        const channelBackup = await lndClient.exportAllChannelBackup();
+        this.writeLndBackup(lndClient.currency, channelBackup);
+
+        this.listenToChannelBackups(lndClient);
+
+        lndClient.subscribeChannelBackups();
+        this.logger.verbose(`Listening to ${currency} LND channel backups`);
+      }
+    }
+  }
+
+  private listenToChannelBackups = (lndClient: LndClient) => {
+    lndClient.on('channelBackup', (channelBackup) => {
+      this.logger.debug(`New ${lndClient.currency} channel backup`);
+      this.writeLndBackup(lndClient.currency, channelBackup);
+    });
+  }
+
+  private startFilewatcher = (client: string, dbPath: string) => {
+    if (fs.existsSync(dbPath)) {
+      let previousDatabaseHash = '';
+
+      this.logger.verbose(`Writing initial ${client} database backup`);
+      const { content, hash } = this.readDatabase(dbPath);
+
+      previousDatabaseHash = hash;
```

Comment:
Minor nit: I think `previousDatabaseHash` can be declared here rather than being initialized above.